compiler/rustc_feature/src/builtin_attrs.rs

```diff
@@ -1004,10 +1069,13 @@ pub fn is_builtin_attr_name(name: Symbol) -> bool {
     BUILTIN_ATTRIBUTE_MAP.get(&name).is_some()
 }
 
-/// Whether this builtin attribute is only used in the local crate.
+/// Whether this builtin attribute is encoded cross crate.
+/// `EncodeCrossCrate::No` means it is only used in the local crate.
 /// If so, it is not encoded in the crate metadata.
-pub fn is_builtin_only_local(name: Symbol) -> bool {
-    BUILTIN_ATTRIBUTE_MAP.get(&name).is_some_and(|attr| attr.only_local)
+pub fn is_builtin_encode_cross_crate(name: Symbol) -> bool {
+    BUILTIN_ATTRIBUTE_MAP.get(&name).is_some_and(|attr| {
```

Comment:
you flip the condition but still use `is_some_and` :thinking: this means that attributes which are *not* builtin now never get encoded in the crate metadata.

Reply:
:sweat_smile: I dislike this function name. It's ambiguous whether it's `!is(builtin) && is(encode_cross_crate)`  or `!is(builtin) || is(encode_cross_crate)` or even `!(is(builtin) && is(encode_cross_crate))`.

I also think we should generally avoid having negation in the name of a function returning a `bool`.

keep it as `fn is_encode_cross_crate` and change it's body to
```rust
pub fn is_encode_cross_crate(name: Symbol) -> bool {
    if let Some(attr) = BUILTIN_ATTRIBUTE_MAP.get(&name) {
        if attr.encode_cross_crate == EncodeCrossCrate::Yes { false } else { true }
    } else {
        true
    }
}
```

Reply:
> 😅 I dislike this function name. It's ambiguous whether it's `!is(builtin) && is(encode_cross_crate)` or `!is(builtin) || is(encode_cross_crate)` or even `!(is(builtin) && is(encode_cross_crate))`.
> 
> I also think we should generally avoid having negation in the name of a function returning a `bool`.
> 
> keep it as `fn is_encode_cross_crate` or even just `fn encode_cross_crate` and change it's body to
> 
> ```rust
> pub fn encode_cross_crate(name: Symbol) -> bool {
>     if let Some(attr) = BUILTIN_ATTRIBUTE_MAP.get(&name) {
>         if attr.encode_cross_crate == EncodeCrossCrate::Yes { false } else { true }
>     } else {
>         true
>     }
> }
> ```

Done. Thank you.